server/app/controllers/HomeController.java

```diff
@@ -46,16 +51,25 @@ public HomeController(
     this.languageUtils = checkNotNull(languageUtils);
     this.faviconURL =
         Optional.ofNullable(Strings.emptyToNull(configuration.getString("whitelabel.favicon_url")));
+    this.featureFlags = featureFlags;
   }
 
   public CompletionStage<Result> index(Http.Request request) {
     Optional<CiviFormProfile> maybeProfile = profileUtils.currentUserProfile(request);
 
+    boolean bypassLogin =
+        featureFlags.getFlagEnabled(request, FeatureFlag.BYPASS_LOGIN_LANGUAGE_SCREENS);
+
+    // If the user isn't already logged in within their browser session, consider them a guest.
     if (maybeProfile.isEmpty()) {
-      return CompletableFuture.completedFuture(
-          redirect(controllers.routes.HomeController.loginForm(Optional.empty())));
+      return bypassLogin
+          ? CompletableFuture.completedFuture(
+              redirect(routes.CallbackController.callback(GuestClient.CLIENT_NAME).url()))
```

Comment:
**issue (blocking):** this will create a new `Account` record every time someone visits the landing page, regardless of whether they continue as a guest or login to their account. Can we delay doing that until they proceed to apply for something as a guest?

Reply:
To clarify, do you mean a new `Applicant` record? Since guests haven't created an `Account` yet?

I need to think about this problem a bit more. It seems to me that having an additional user mode other than "logged in" and "guest" would add a significant amount of complexity to the rest of the application. For example, in many places we check whether a user "is a guest" and assume that if not, they aren't logged in (header bar is an instance of this).

While it's true that visiting the landing page as a new user creates a guest record for them, refreshing the page or revisiting it later doesn't (IIUC). Also, I anecdotally suspect that most guests will click "Apply" on **something**, if not out of sheer curiosity to see what happens, and that would render this optimization useless.

How significant of a problem is this in your opinion to warrant adding this new user state? Am I missing something?

Reply:
> Also, I anecdotally suspect that most guests will click "Apply" on something, if not out of sheer curiosity to see what happens, and that would render this optimization useless.

@gwendolyngoetz @elisekalstad would it be possible to check Seattle GA to get a sense of unique monthly visitors to the landing page vs unique monthly visitors to the sum of all the `/programs/<programs-slug>` routes?

Reply:
discussed out-of-band in a meeting, good to proceed